### DISCUSSION TITLE
Is SYMBOL-MACROLET supported?

### DISCUSSION THREAD

**Discussion by pamoroso** [Interlisp/medley#1643] [General]
Does Medley Common Lisp support `symbol-macrolet`? The only relevant symbol I can find is internal in the `LOOP` package where the only references in [sources/XCL-LOOP](https://github.com/Interlisp/medley/blob/master/sources/XCL-LOOP), [here](https://github.com/Interlisp/medley/blob/f6eb5d984667d1b0846c19c33341880a5832c720/sources/XCL-LOOP#L296) and [here](https://github.com/Interlisp/medley/blob/f6eb5d984667d1b0846c19c33341880a5832c720/sources/XCL-LOOP#L1304), don't define or implement it.

Here is a test session in an XCL Exec in Medley Online (19-Mar-2024 02:06:49). The code snippets are from the HyperSpec.

```lisp
2/8> (apropos "SYMBOL-MACROLET")
LOOP::SYMBOL-MACROLET
SYMBOL-MACROLET
IL:|{DSK}<home>medley>il>demo>SYMBOL-MACROLET.DRIBBLE;1|
SYMBOL-MACROLET
2/9> (symbol-macrolet ((x 'foo)) 
   (list x (let ((x 'bar)) x)))
Undefined car of form
SYMBOL-MACROLET

2/10> 

2/10> (symbol-macrolet ((x '(foo x))) 
   (list x))
Undefined car of form
SYMBOL-MACROLET

2/11>
```

There's another `symbol-macrolet` reference (in `XCL-USER`?) with no definition. This is a fresh `XCL-USER` session where I haven't evaluated the symbol `symbol-macrolet` prior to the above snippets.

**Comment by tfeb** [comment]
It would be interesting to see what happens when PCL is loaded, if it's not there by default.  One of the motivations for `symbol-macrolet` was supporting things like `with-slots`, for instance.

My memory is too dim (and I am ashamed to say I don't use Medley very often), but I would not be at all surprised if PCL didn't have some hack to make it work.  Perhaps not, though: it might have done it by code-walking things.

**Comment by pamoroso** [comment]
As far as I know PCL is loaded by default in Medley. It certainly was in the image I ran the tests in.

`symbol-macrolet` is indeed missing from Medley. The code in sources/XCL-LOOP is just a stub @masinter added to make Sacla `loop` work.

By the way we may turn this discussion into an issue to record the missing feature.
